### ISSUE TITLE
Should chaos mesh support modifying chaos spec?

### ISSUE THREAD

**Issue by fewdan** [chaos-mesh/chaos-mesh#2227]
## Feature Request

**Is your feature request related to a problem? Please describe:**

Because of #1905 , the current chaos mesh does not support modifying the chaos spec.

**Describe the feature you'd like:**

I think it is necessary to modify the spec when running chaos. But will this cause any problems? 

@YangKeao Any good ideas?

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

**Comment by g1eny0ung** [comment]
After our @chaos-mesh/maintainers @chaos-mesh/committers discussion,  **TL;DR, even though we used to support this feature, we have decided not to support it for now.**

In general, the scope of allowed changes is not well defined. Should we allow users to change all the features? Obviously not, if you update the name of chaos, you create new chaos, or if we change the action of a NetworkChaos, it will be completely different chaos!

At the same time, to meet the limited scope updates, we need to extend our validation, which will increase the complexity of the code to some extent. While this is not impossible to do, it is not the solution we expect, it is like patching all the time. In the future, we are bound to introduce new chaos, and then we have to continue patching again. 😵

So, in the case that we can't clearly grasp the behavior generated by the definition, we will discuss this feature for the time being. If in the future we find a better solution or a strong user need arises, we will continue to revisit this issue.

**Comment by mladedav** [comment]
And if there is a need or we want to test different values of chaos spec, what is the preferred solution? Create multiple chaos specs and pause/resume them? Delete old specs and create new ones?

**Comment by fewdan** [comment]
> And if there is a need or we want to test different values of chaos spec, what is the preferred solution? Create multiple chaos specs and pause/resume them? Delete old specs and create new ones?

Yes, I think the current plan is: create chaos with different specs, and compare these chaos.
